### PR TITLE
[SPARK-33775][BUILD] Suppress sbt compilation warnings in Scala 2.13 

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
There are too many compilation warnings in Scala 2.13, this pr add some `-Wconf:msg= regexes` rules to `SparkBuild.scala` to suppress compilation warnings and the suppressed will not be printed to the console.

The suppressed compilation warnings includes:

- All warnings related to `method\value\type\object\trait\inheritance` deprecated since 2.13

- All warnings related to `Widening conversion from XXX to YYY is deprecated because it loses precision`

- Auto-application to `()` is deprecated. Supply the empty argument list `()` explicitly to invoke method `methodName`, or remove the empty argument list from its definition (Java-defined methods are exempt).In Scala 3, an unapplied method like this will be eta-expanded into a function.

- method with a single empty parameter list overrides method without any parameter list

- method without a parameter list overrides a method with a single empty one

Not suppressed compilation warnings includes:

- Unicode escapes in triple quoted strings are deprecated, use the literal character instead.

- view bounds are deprecated

- symbol literal is deprecated

### Why are the changes needed?
Suppress unimportant compilation warnings in Scala 2.13

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass the Jenkins or GitHub Action

### CODE REVIEW

**Comment by srowen** [comment]
Does this just suppress for one file?

**Comment by LuciferYang** [comment]
> Does this just suppress for one file?

Will suppress all files use `-Wconf:msg= regexes`

**Comment by LuciferYang** [comment]
@srowen It seems that all compilation warnings added in Scala 2.13 have been suppressed,

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -238,7 +238,16 @@ object SparkBuild extends PomBuild {
           "-Wconf:cat=other-match-analysis&site=org.apache.spark.sql.catalyst.catalog.SessionCatalog.lookupFunction.catalogFunction:wv",
           "-Wconf:cat=other-pure-statement&site=org.apache.spark.streaming.util.FileBasedWriteAheadLog.readAll.readFile:wv",
           "-Wconf:cat=other-pure-statement&site=org.apache.spark.scheduler.OutputCommitCoordinatorSuite.<local OutputCommitCoordinatorSuite>.futureAction:wv",
-          "-Wconf:cat=other-pure-statement&site=org.apache.spark.sql.streaming.sources.StreamingDataSourceV2Suite.testPositiveCase.\\$anonfun:wv"
+          "-Wconf:cat=other-pure-statement&site=org.apache.spark.sql.streaming.sources.StreamingDataSourceV2Suite.testPositiveCase.\\$anonfun:wv",
+          // SPARK-33775 Suppress compilation warnings that contain the following contents.
+          "-Wconf:msg=^(?=.*?method|value|type|object|trait|inheritance)(?=.*?deprecated)(?=.*?since 2.13).+$:s",
+          "-Wconf:msg=^(?=.*?Widening conversion from)(?=.*?is deprecated because it loses precision).+$:s",
+          "-Wconf:msg=Auto-application to \\`\\(\\)\\` is deprecated:s",
+          "-Wconf:msg=Unicode escapes in triple quoted strings are deprecated:s",
```
Comment (srowen) [inline project/SparkBuild.scala:246]: Some of these we should maybe just fix, like how many unicode escape issues do we have?
We've already got rid of most of the symbol literal usages and view bounds.
The rest look OK, maybe better to suppress than fix. We have to remember to undo some of them eventually like paying attention to what's deprecated in 2.13.

Comment (LuciferYang) [reply project/SparkBuild.scala]: There are total 15 compilation warnings related to `Unicode escapes in triple quoted strings are deprecated` in core and sql module.

Address 8e03c13 remove this match, and SPARK-33801 use to track this issue

Comment (LuciferYang) [reply project/SparkBuild.scala]: There are only 3 compilation warnings related to `view bounds are deprecated` in `SequenceFileRDDFunctions` , Address d97e0a7 remove this match, and SPARK-33804 use to track this issue

Comment (LuciferYang) [reply project/SparkBuild.scala]: Address 3cd8174 remove msg match of `symbol literal is deprecated` and left TODO(SPARK-33805) to remind undo the corresponding deprecated usage suppression rule after fixed.

**Review by srowen — APPROVED** [review]
Seems OK for now to make the compiler output more usable to track down the warnings we can more readily fix in the short term.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thanks, @LuciferYang and @srowen .
Merged to master/3.1.

**Comment by LuciferYang** [comment]
thx ~ @srowen and @dongjoon-hyun

**Pull request closed without merging**